src/experimental/tools/download2go/api/drm/keySystems.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright 2019 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IDBPDatabase } from "idb";
+import { map } from "rxjs/operators";
+
+import EMEManager from "../../../../../core/eme/eme_manager";
+import {
+  IKeySystemOption,
+  IPersistedSessionData,
+} from "../../../../../core/eme/types";
+import { IndexDBError } from "../../utils";
+import { IUtilsKeySystemsTransaction } from "./types";
+
+/**
+ * Get the licence when keysSystems are specified
+ *
+ * @remarks
+ * This function is basically reproducing the getLicence from the rx-player, but we are
+ * adding an additional step to catch the licence and resolve a promise with the licence
+ * To get the challenge we need to retrieve the licence
+ * we are instanciating a minimal rxPlayer
+ * @param ILicenceOptions - The parameters we need to get the licence
+ * @returns The licence under a buffer form
+ *
+ */
+function EMETransaction(
+  KeySystemsOption: IKeySystemOption,
+  keySystemsUtils: IUtilsKeySystemsTransaction,
+  db: IDBPDatabase
+) {
+  const video = document.createElement("video");
+  const { contentID, contentProtection$ } = keySystemsUtils;
+  const keySystems = [
+    {
+      ...KeySystemsOption,
+      licenseStorage: {
+        save(sessionsIDS: IPersistedSessionData[]) {
+          db.add("contentsProtection", {
+            contentID,
+            drmKey: `${contentID}--${Date.now()}`,
```

Comment:
Are you sure about that?

Reply:
I can see many problems with using Date.now as an ID:
  1. what if the user's pc is really fast
  2. what if the user's clock is updated
  3. what about leap seconds

Those are very rare issues but still, using Date.now for this use case is surely not the best way to do it.

Reply:
I understand I agree that is not the best way to do it!

I added a fix for this, we are now using a simple counter at the top level.